script.js

```diff
@@ -121,15 +128,42 @@ const markDoneTodoHandler = (
 
     todo.title = sanitizeInput(inputElement.value).trim();
     todo.isCompleted = true;
+    filterTodosHandler(filterValue);
 };
 
 const searchHandler = () => {
     const searchedValue = $searchInput.value.toLowerCase().trim();
-    const searchedArray = todos.filter((todo) =>
+    searchedArray = todos.filter((todo) =>
         todo.title.toLowerCase().includes(searchedValue)
     );
 
-    renderTodos(searchedArray);
+    filterTodosHandler(filterValue);
+};
+
+const filterTodosHandler = (toFilterValue) => {
+    isSearched = $searchInput.value.trim().length ? true : false;
+    let tobeFilteredArray = isSearched ? searchedArray : todos;
+
+    switch (toFilterValue) {
+        case "incomplete":
+            filteredArray = tobeFilteredArray.filter(
+                (todo) => !todo.isCompleted
+            );
+            break;
+
+        case "complete":
+            filteredArray = tobeFilteredArray.filter(
+                (todo) => todo.isCompleted
+            );
+            break;
+
+        default:
+            filteredArray = [...tobeFilteredArray];
+            break;
+    }
```

Comment:
```suggestion
  let filteredArray = [...tobeFilteredArray];

  if (toFilterValue === "INCOMPLETE") {
    filteredArray = tobeFilteredArray.filter((todo) => !todo.isCompleted);
  }

  if (toFilterValue === "COMPLETE") {
    filteredArray = tobeFilteredArray.filter((todo) => todo.isCompleted);
  }
```

Reply:
The conditions is based on same filter state parameter only and there is no other additional conditions required. So isn't it better and looking goods to use "Switch Case" not using  "if else statement"?